storage/innobase/include/row0upd.h

```diff
@@ -568,6 +568,8 @@ struct upd_node_t{
 	dtuple_t*	row;	/*!< NULL, or a copy (also fields copied to
 				heap) of the row to update; this must be reset
 				to NULL after a successful update */
+	dtuple_t*	historical_row;	/*!< historical row used in
+				CASCADE UPDATE/SET NULL */
```

Comment:
The comment should say where the memory for `historical_row` is allocated from, and what its lifetime is (that is, when it is assigned non-`NULL` and back to `NULL`).